### PR TITLE
Custom regex escape

### PR DESCRIPTION
**Pull request by tahmidefaz**
## Overview

This PR is being created to address [this Jira](https://issues.redhat.com/browse/RHCLOUD-10860).
This PR provides a custom regex escape method in HBI.

## Secure Coding Practices Checklist GitHub Link

- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist

- [x] Input Validation
- [x] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by kruai — APPROVED** [review]
I think this looks good, but I just want to make sure... we only need to modify the `re.escape()` calls, right? I would assume we don't need to worry about our calls to `re.findall()`, `re.match()`, and `re.fullmatch()`

**Comment by tahmidefaz** [comment]
> I think this looks good, but I just want to make sure... we only need to modify the `re.escape()` calls, right? I would assume we don't need to worry about our calls to `re.findall()`, `re.match()`, and `re.fullmatch()`

Yes, that's correct! Just the `re.escape()` calls

File: tests/test_unit.py [hunk 1]
```diff
@@ -1984,5 +1985,18 @@ def test_custom_fields_parser(self):
...
+class CustomRegexMethodTestCase(TestCase):
+    def test_custom_regex_escape(self):
+        for regex_input, output in (
+            (".?well+", "\\.\\?well\\+"),
+            ("&[^abc]~", "\\&\\[^abc\\]\\~"),
+            ("some|*#thing", "some\\|\\*\\#thing"),
+            ('.?+*|{}[]()"\\#@&<>~', '\\.\\?\\+\\*\\|\\{\\}\\[\\]\\(\\)\\"\\\\\\#\\@\\&\\<\\>\\~'),
+            ("\\", "\\\\"),
+        ):
+            result = custom_escape(regex_input)
```
Comment (ryandillinfelton) [inline tests/test_unit.py:1997]: This works fine as is, but i think the cause of the test failure is easier to track down when you wrap each assert in a subtest. It gives the the test case it failed on in the header which I find to be helpful.
```suggestion
        with self.subTest(regex_input=regex_input):
            result = custom_escape(regex_input)
```

**Review by ryandillinfelton — DISMISSED** [review]
Looks good. I'm cool with merging it as is but left a simple suggestion for a small test improvement. LMK if you choose to use the suggestion and I can re-approve quick :t-rex:

File: tests/test_unit.py (hunk 1 above)
Comment (tahmidefaz) [reply tests/test_unit.py]: done! thanks! :+1:

**Review by ryandillinfelton — APPROVED** [review]
Looks :goat: to me!

**Review by thearifismail — APPROVED** [review]
LGTM.

**Pull request merged**